include/cantera/kinetics/InterfaceRate.h

```diff
@@ -385,6 +385,9 @@ class InterfaceRate : public RateType, public InterfaceRateBase
     {
         InterfaceRateBase::setParameters(node);
         RateType::setParameters(node, rate_units);
+        if (node.hasKey("coverage-dependencies")) {
+            RateType::setCompositionDependence(true);
+        }
```

Comment:
The place where `setCompositionDependence` needs to be called is from `addCoverageDependence`, which is a method of `InterfaceRateBase`. You may need to add an override of `addCoverageDependence` in this class to do that.